Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp

```diff
@@ -0,0 +1,255 @@
+using namespace Windows::Storage;
+using namespace Windows::Graphics::Imaging;
+using namespace Windows::Media::Core;
+IVectorView<MediaFrameSourceGroup> sourceGroups;
+// Iterates through source groups and filter-out the frame sources which have IR,depth and other sources which we cannot consume in this app
+IVector<MediaFrameSourceInfo> GetFilteredSourceGroupList()
+{
+    auto filteredSourceInfos = single_threaded_vector<MediaFrameSourceInfo>();
+    sourceGroups = MediaFrameSourceGroup::FindAllAsync().get();
+    auto sourceGroupIter = sourceGroups.First();
+    while (sourceGroupIter.HasCurrent())
+    {
+        std::wcout << sourceGroupIter.Current().DisplayName().c_str();
+        auto sourceInfos = sourceGroupIter.Current().SourceInfos();
+        auto sourceInfoIter = sourceInfos.First();
+        // iterate through sources and filter-out the IR,depth and other sources which we cannot consume in this app
+        while (sourceInfoIter.HasCurrent())
+        {
+            if ((sourceInfoIter.Current().MediaStreamType() == MediaStreamType::Photo
+                || sourceInfoIter.Current().MediaStreamType() == MediaStreamType::VideoPreview
+                || sourceInfoIter.Current().MediaStreamType() == MediaStreamType::VideoRecord)
+                && sourceInfoIter.Current().SourceKind() == MediaFrameSourceKind::Color)
+            {
+                filteredSourceInfos.Append(sourceInfoIter.Current());
+            }
+            sourceInfoIter.MoveNext();
+        }
+        sourceGroupIter.MoveNext();
+    }
+    return filteredSourceInfos;
+}
+
+int GetSGSelection(IVector<MediaFrameSourceInfo> filteredGroup)
+{
+    unsigned int idx = 0;
+    auto group = filteredGroup.First();
+    while (group.HasCurrent())
+    {
+        idx++;
+        auto currGroup = group.Current();
+        if (currGroup.SourceGroup() == nullptr)
+        {
+            std::cout << "nullptr";
+        }
+        // These are in the same order as the enum MediaStreamType
+        // TODO: better solution is to create a Pair type lookup which is future proof
+        std::wstring streamTypes[] = 
+        { 
+            L"VideoPreview", 
+            L"VideoRecord", 
+            L"Audio", 
+            L"Photo" 
+        };
+
+        // These are in the same order as the enum Panel
+        std::wstring panelTypes[] = 
+        {
+            L"Unknown",
+            L"Front",
+            L"Back",
+            L"Top",
+            L"Bottom",
+            L"Left",
+            L"Right"
+        };
+        auto enclosureLocation = currGroup.DeviceInformation().EnclosureLocation();
+        std::wstring panelLocation;
+        if (enclosureLocation)
+        {
+            panelLocation = panelTypes[(int)enclosureLocation.Panel()];
+        }
+        
+        std::wcout << idx << L":" << currGroup.DeviceInformation().Name().c_str() << ":" << streamTypes[(int)currGroup.MediaStreamType()].c_str() << panelLocation.c_str() << std::endl;

```

Comment:
>wcout [](start = 13, length = 5)

why not use cout and string throughout instead of both wstring and string, wcout and cout